pkg/fixtures/writer_s3.go

```diff
@@ -0,0 +1,108 @@
+package fixtures
+
+import (
+	"context"
+	"github.com/applike/gosoline/pkg/blob"
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/aws/aws-sdk-go/aws"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type BlobFixturesSettings struct {
+	BasePath   string
+	ConfigName string
+}
+
+type blobFixtureWriter struct {
+	logger      mon.Logger
+	batchRunner *blob.BatchRunner
+	purger      *blobPurger
+	settings    *BlobFixturesSettings
+	store       blob.Store
+}
+
+func BlobFixtureWriterFactory(settings *BlobFixturesSettings) FixtureWriterFactory {
+	return func(config cfg.Config, logger mon.Logger) (FixtureWriter, error) {
+		store := blob.NewStore(config, logger, settings.ConfigName)
+		br := blob.NewBatchRunner(config, logger)
+		purger := newBlobPurger(config, logger, settings)
+
+		return NewBlobFixtureWriterWithInterfaces(logger, store, br, settings, purger), nil
+	}
+}
+
+func NewBlobFixtureWriterWithInterfaces(logger mon.Logger, store blob.Store, batchRunner *blob.BatchRunner, settings *BlobFixturesSettings, purger *blobPurger) FixtureWriter {
+	return &blobFixtureWriter{
+		logger:      logger,
+		store:       store,
+		settings:    settings,
+		batchRunner: batchRunner,
+		purger:      purger,
+	}
+}
+
+func (s *blobFixtureWriter) Purge() error {
+	return s.purger.purge()
+}
+
+func (s *blobFixtureWriter) Write(*FixtureSet) error {
+	s.store.CreateBucket()
+
+	var files []string
+	err := filepath.Walk(s.settings.BasePath, func(path string, f os.FileInfo, err error) error {
+		if f.IsDir() {
+			return nil
+		}
+
+		files = append(files, path)
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	replaceStr := ""
+	if s.settings.BasePath != "." && s.settings.BasePath != "./" {
+		replaceStr = s.settings.BasePath
+	}
+
+	var batch blob.Batch
+	for _, file := range files {
+		reader, err := os.Open(file)
+		if err != nil {
+			return err
+		}
+
+		fileInfo, _ := reader.Stat()
+		size := fileInfo.Size()
+		buffer := make([]byte, size)
+		_, err = reader.Read(buffer)
+		if err != nil {
+			return err
+		}
```

Comment:
in one of the older discussions i told you to use ioutil.ReadFile, because you don't need to check 3 errors (of which you actually ignored one)

Reply:
Already applied.